From 57d9a630987e6cfb6b97f645398a331a715a1156 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 11 Oct 2024 10:40:09 +0200
Subject: [PATCH 1288/1695] drivers: stm32_remote_proc: add
 stm32_rproc_is_secure()

Add stm32_remoteproc driver API function stm32_rproc_is_secure()
that return whether of not remote processor management shall be
handled through OP-TEE remoteproc secure services.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git bc04c5cf83819217f3b6a688303d84c855dbdbab drivers: stm32_remote_proc: add stm32_rproc_is_secure()]
Change-Id: I8363b8e2fffe6f6e8c5278032c069c329e217a58
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/410657
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 10 ++++++++++
 core/include/drivers/stm32_remoteproc.h    | 10 ++++++++++
 2 files changed, 20 insertions(+)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index 8a339a3a3..80199d9da 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -129,6 +129,16 @@ void *stm32_rproc_get(uint32_t rproc_id)
 	return rproc;
 }
 
+bool stm32_rproc_is_secure(uint32_t rproc_id)
+{
+	struct stm32_rproc_instance *rproc = stm32_rproc_get(rproc_id);
+
+	if (rproc)
+		return !rproc->cdata->ns_loading;
+
+	return false;
+}
+
 /* Re-apply default access right on the memory regions */
 static TEE_Result
 stm32_rproc_release_mems_access(struct stm32_rproc_instance *rproc)
diff --git a/core/include/drivers/stm32_remoteproc.h b/core/include/drivers/stm32_remoteproc.h
index df6eda311..70fceb1c1 100644
--- a/core/include/drivers/stm32_remoteproc.h
+++ b/core/include/drivers/stm32_remoteproc.h
@@ -104,4 +104,14 @@ TEE_Result stm32_rproc_get_mem(uint32_t rproc_id);
  */
 TEE_Result stm32_rproc_release_mem(uint32_t rproc_id);
 
+#ifdef CFG_STM32MP_REMOTEPROC
+/* Return true is secure loading of remoteproc firmware is enabled */
+bool stm32_rproc_is_secure(uint32_t rproc_id);
+#else
+static inline bool stm32_rproc_is_secure(uint32_t rproc_id __unused)
+{
+	return false;
+}
+#endif
+
 #endif /* __DRIVERS_STM32_REMOTEPROC_H */
-- 
2.47.2

